youtube_dl/downloader/external.py

```diff
@@ -270,6 +271,10 @@ def _call_downloader(self, tmpfilename, info_dict):
                 args += ['-rtmp_live', 'live']
 
         args += ['-i', url, '-c', 'copy']
+
+        if self.params.get('test', False):
+            args += ['-fs', str(self._TEST_FILE_SIZE)] # -fs limit_size (output), expressed in bytes
```

Comment:
`compat_str` most likely appeared after code in `http.py` was written. In general this matters for passing arguments to subprocess. Anyway this is inconsistent when some of arguments are strings and another are bytestrings.